tests/test_interdiff_command.rs

```diff
@@ -0,0 +1,151 @@
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file2"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "left"]);
+
+    test_env.jj_cmd_success(&repo_path, &["checkout", "root"]);
+    std::fs::write(repo_path.join("file3"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file2"), "foo\nbar\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "right"]);
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &["interdiff", "--from", "left", "--to", "right"],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    Modified regular file file2:
+       1    1: foo
+            2: bar
+    "###);
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &["interdiff", "--from", "left", "--to", "right", "-s"],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    M file2
+    "###);
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &["interdiff", "--from", "left", "--to", "right", "--git"],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    diff --git a/file2 b/file2
+    index 257cc5642c...3bd1f0e297 100644
+    --- a/file2
+    +++ b/file2
+    @@ -1,1 +1,2 @@
+     foo
+    +bar
+    "###);
+}
+
+#[test]
+fn test_interdiff_paths() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "foo\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file1"), "bar\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "bar\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "left"]);
+
+    test_env.jj_cmd_success(&repo_path, &["checkout", "root"]);
+    std::fs::write(repo_path.join("file1"), "foo\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file1"), "baz\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "baz\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "right"]);
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &["interdiff", "--from", "left", "--to", "right", "file1"],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    Modified regular file file1:
+       1    1: barbaz
+    "###);
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &[
+            "interdiff",
+            "--from",
+            "left",
+            "--to",
+            "right",
```

Comment:
nit: You could leave out `--to right` in most cases since `@` is the default. That makes the tests a bit shorter (especially this one) and we get to test that the default works as expected. You can also skip creating the `right` branch if you don't use it.

I would probably just leave one case with `--to` where its argument is not the same commit as `@` if we want to check that the argument is interpreted correctly.

Reply:
I ended up doing it in reverse; most use `--right` and one defaults to `@`. As a matter of taste, I like seeing explicit arguments in tests so that I can think less about the test setup.